driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverFactory.java

```diff
@@ -149,34 +149,34 @@ public WebDriverRequest prepareWebDriverRequest(WebDriverRequest webDriverReques
     }
 
     @Override
-    public void setupNewWebDriverSession(EventFiringWebDriver eventFiringWebDriver, SessionContext sessionContext) {
+    public WebDriver setupNewWebDriverSession(WebDriver webDriver, SessionContext sessionContext) {
 
         DesktopWebDriverRequest desktopWebDriverRequest = (DesktopWebDriverRequest) sessionContext.getWebDriverRequest();
         final String browser = desktopWebDriverRequest.getBrowser();
 
-        // add event listeners
-        if (Testerra.Properties.DEMO_MODE.asBool()) {
-            eventFiringWebDriver.register(new VisualEventDriverListener());
-        }
-
-        eventFiringWebDriver.register(new EventLoggingEventDriverListener());
-
         if (!desktopWebDriverRequest.getBaseUrl().isPresent()) {
             WebDriverManager.getConfig().getBaseUrl().ifPresent(desktopWebDriverRequest::setBaseUrl);
         }
+        VisualEventDriverListener visualListener = new VisualEventDriverListener();
```

Comment:
Why not instantiating `VisualEventDriverListener` only when `Testerra.Properties.DEMO_MODE.asBool()`?